# Data Wrangling:
### For Analysis such as: ANOVAS, MANOVAS, Correlation, Chi- Square, Random Forest, Regressions
### For Tableau 

In [3]:
#Table of Content: 

#England and UK: 
## 1st Data Wrangle: Sepreated (total and type HazNonHaz) & Merged (Mergining Classificationa and Treatment Datasets)
## 2nd Data Wrangle: Updated (removing unwanted columns, and renaming columns)
## 3rd Data Wrangle: Reshaping 

In [4]:
# Import Packages

In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
pd.options.display.max_rows = None
pd.options.display.max_columns = None

pd.set_option('display.max_rows', 500)

from itertools import product

# 1st Data Wrangle 

In [6]:
# devins starter code

## Import Data

In [7]:
# Datasets were wrangled in R

In [8]:
# Updated_England_Industry
# Updated_England_Management
# Updated_Uk_Industry
# Updated_Uk_Management

# *England & UK*

In [9]:
# Total (without Haz/NonHaz Split) and Type (with Haz/NonHaz Split)

# Data Wrangle - Part 1

In [10]:
# iloc -> index location (calling only the first column - column 0)

In [11]:
england_industry_df = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/Updated_England_Industry.1.csv").iloc[:,1:]
england_managment_df = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/Updated_England_Management.1.csv").iloc[:,1:]

In [12]:
uk_industry_df = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/Updated_UK_Industry.1.csv").iloc[:,1:]
uk_managment_df = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/Updated_UK_Management.1.csv").iloc[:,1:]

In [13]:
# Calling the column names of the datasets
# Take a closer look at the columns of each dataset and see if anything is repeated, missing, or not neccessary

In [14]:
england_industry_df.columns

Index(['Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'A', 'B', 'C10-C12', 'C13-C15', 'C16',
       'C17_C18', 'C19', 'C20-C22', 'C23', 'C24_C25', 'C26-C30', 'C31-C33',
       'D', 'E36_E37_E39', 'F', 'G-U_X_G4677', 'EP_HH',
       'Total waste generation'],
      dtype='object')

In [15]:
england_managment_df.columns

Index(['Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'Energy recovery', 'Incineration',
       'Recovery other than energy recovery - Except backfilling',
       'Recovery other than energy recovery - Backfilling',
       'Deposit onto or into land',
       'Land treatment and release into water bodies', 'd`'],
      dtype='object')

In [16]:
uk_industry_df.columns

Index(['Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'A', 'B', 'C10-C12', 'C13-C15', 'C16',
       'C17_C18', 'C19', 'C20-C22', 'C23', 'C24_C25', 'C26-C30', 'C31-C33',
       'D', 'E36_E37_E39', 'F', 'G-U_X_G4677', 'EP_HH',
       'Total waste generation'],
      dtype='object')

In [17]:
uk_managment_df.columns

Index(['Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'Energy recovery', 'Incineration',
       'Recovery other than energy recovery - Except backfilling',
       'Recovery other than energy recovery - Backfilling',
       'Deposit onto or into land',
       'Land treatment and release into water bodies',
       'Total waste sent to final treatment'],
      dtype='object')

# Data Wrangle -> Part 2

In [18]:
# in order to anaylize the data some things need to be changed... bc... for instance...
# loc -> location 
# calling the dataset.loc (in that df we only want where H/NH split == and != Total)

In [19]:
# Spliting Total and Haz/Non-Haz into two different dataframes of Industries Waste Gen

In [20]:
england_industry_total = england_industry_df.loc[england_industry_df['Hazardous/Non-hazardous split'] == "Total"]
england_industry_type = england_industry_df.loc[england_industry_df['Hazardous/Non-hazardous split'] != "Total"]

In [21]:
# Spliting Total and Haz/Non-Haz into two different dataframes of Industries Waste Gen

In [22]:
england_managment_total = england_managment_df.loc[england_managment_df['Hazardous/Non-hazardous split'] == "Total"]
england_managment_type = england_managment_df.loc[england_managment_df['Hazardous/Non-hazardous split'] != "Total"]

In [23]:
# Spliting Total and Haz/Non-Haz into two different dataframes of Industries Waste Gen

In [24]:
uk_industry_total = uk_industry_df.loc[uk_industry_df['Hazardous/Non-hazardous split'] == "Total"]
uk_industry_type = uk_industry_df.loc[uk_industry_df['Hazardous/Non-hazardous split'] != "Total"]

In [25]:
# Spliting Total and Haz/Non-Haz into two different dataframes of Industries Waste Gen

In [26]:
uk_managment_total = uk_managment_df.loc[uk_managment_df['Hazardous/Non-hazardous split'] == "Total"]
uk_managment_type = uk_managment_df.loc[uk_managment_df['Hazardous/Non-hazardous split'] != "Total"]

In [27]:
# double checking one of the above dataframe to see if the code above worked

In [28]:
england_managment_total

Year     EWC-STAT code  \
2    2010              01.1   
5    2010              01.2   
8    2010              01.3   
11   2010    01.4, 02, 03.1   
14   2010              03.2   
17   2010              03.3   
20   2010                05   
23   2010              06.1   
26   2010              06.2   
29   2010              06.3   
32   2010              07.1   
35   2010              07.2   
38   2010              07.3   
41   2010              07.4   
44   2010              07.5   
47   2010              07.6   
50   2010              07.7   
53   2010               08*   
56   2010              08.1   
59   2010             08.41   
62   2010              09.1   
65   2010              09.2   
68   2010              09.3   
71   2010              10.1   
74   2010              10.2   
77   2010              10.3   
80   2010                11   
83   2010              12.1   
86   2010  12.2, 12.3, 12.5   
89   2010              12.4   
92   2010              12.6   
95   2010              12.7   
98   2010          12.8, 13   
101  2010             Total   
104  2012              01.1   
107  2012              01.2   
110  2012              01.3   
113  2012    01.4, 02, 03.1   
116  2012              03.2   
119  2012              03.3   
122  2012                05   
125  2012              06.1   
128  2012              06.2   
131  2012              06.3   
134  2012              07.1   
137  2012              07.2   
140  2012              07.3   
143  2012              07.4   
146  2012              07.5   
149  2012              07.6   
152  2012              07.7   
155  2012               08*   
158  2012              08.1   
161  2012             08.41   
164  2012              09.1   
167  2012              09.2   
170  2012              09.3   
173  2012              10.1   
176  2012              10.2   
179  2012              10.3   
182  2012                11   
185  2012              12.1   
188  2012  12.2, 12.3, 12.5   
191  2012              12.4   
194  2012              12.6   
197  2012              12.7   
200  2012          12.8, 13   
203  2012             Total   
206  2014              01.1   
209  2014              01.2   
212  2014              01.3   
215  2014    01.4, 02, 03.1   
218  2014              03.2   
221  2014              03.3   
224  2014                05   
227  2014              06.1   
230  2014              06.2   
233  2014              06.3   
236  2014              07.1   
239  2014              07.2   
242  2014              07.3   
245  2014              07.4   
248  2014              07.5   
251  2014              07.6   
254  2014              07.7   
257  2014               08*   
260  2014              08.1   
263  2014             08.41   
266  2014              09.1   
269  2014              09.2   
272  2014              09.3   
275  2014              10.1   
278  2014              10.2   
281  2014              10.3   
284  2014                11   
287  2014              12.1   
290  2014  12.2, 12.3, 12.5   
293  2014              12.4   
296  2014              12.6   
299  2014              12.7   
302  2014          12.8, 13   
305  2014             Total   
308  2016              01.1   
311  2016              01.2   
314  2016              01.3   
317  2016    01.4, 02, 03.1   
320  2016              03.2   
323  2016              03.3   
326  2016                05   
329  2016              06.1   
332  2016              06.2   
335  2016              06.3   
338  2016              07.1   
341  2016              07.2   
344  2016              07.3   
347  2016              07.4   
350  2016              07.5   
353  2016              07.6   
356  2016              07.7   
359  2016               08*   
362  2016              08.1   
365  2016             08.41   
368  2016              09.1   
371  2016              09.2   
374  2016              09.3   
377  2016              10.1   
380  2016              10.2   
383  2016              10.3   
386  2

In [29]:
# Combining two subject indentifier -> Year and EWC-STAT description -> which will make each row unique

# - They did not match on the rows that were aggregates "Total waste generated" and "Total waste managed"

# - So we removed those because they were rows that represented total's

# Line 2: double checking to see what are false 

In [30]:
combine_yr_ewc = england_industry_total[['Year', 'EWC-STAT description']] == england_managment_total[['Year', 'EWC-STAT description']]
combine_yr_ewc

Year  EWC-STAT description
2    True                  True
5    True                  True
8    True                  True
11   True                  True
14   True                  True
17   True                  True
20   True                  True
23   True                  True
26   True                  True
29   True                  True
32   True                  True
35   True                  True
38   True                  True
41   True                  True
44   True                  True
47   True                  True
50   True                  True
53   True                  True
56   True                  True
59   True                  True
62   True                  True
65   True                  True
68   True                  True
71   True                  True
74   True                  True
77   True                  True
80   True                  True
83   True                  True
86   True                  True
89   True                  True
92   True                  True
95   True                  True
98   True                  True
101  True                 False
104  True                  True
107  True                  True
110  True                  True
113  True                  True
116  True                  True
119  True                  True
122  True                  True
125  True                  True
128  True                  True
131  True                  True
134  True                  True
137  True                  True
140  True                  True
143  True                  True
146  True                  True
149  True                  True
152  True                  True
155  True                  True
158  True                  True
161  True                  True
164  True                  True
167  True                  True
170  True                  True
173  True                  True
176  True                  True
179  True                  True
182  True                  True
185  True                  True
188  True                  True
191  True                  True
194  True                  True
197  True                  True
200  True                  True
203  True                 False
206  True                  True
209  True                  True
212  True                  True
215  True                  True
218  True                  True
221  True                  True
224  True                  True
227  True                  True
230  True                  True
233  True                  True
236  True                  True
239  True                  True
242  True                  True
245  True                  True
248  True                  True
251  True                  True
254  True                  True
257  True                  True
260  True                  True
263  True                  True
266  True                  True
269  True                  True
272  True                  True
275  True                  True
278  True                  True
281  True                  True
284  True                  True
287  True                  True
290  True                  True
293  True                  True
296  True                  True
299  True                  True
302  True                  True
305  True                 False
308  True                  True
311  True                  True
314  True                  True
317  True                  True
320  True                  True
323  True                  True
326  True                  True
329  True                  True
332  True                  True
335  True                  True
338  True                  True
341  True                  True
344  True                  True
347  True                  True
350  True                  True
353  True                  True
356  True                  True
359  True                  True
362  True                  True
365  True                  True
368  True                  True
371  True                  True
374  

In [31]:
combine_yr_ewc2 = england_industry_type[['Year', 'EWC-STAT description']] == england_managment_type[['Year', 'EWC-STAT description']]
combine_yr_ewc2

Year  EWC-STAT description
0    True                  True
1    True                  True
3    True                  True
4    True                  True
6    True                  True
7    True                  True
9    True                  True
10   True                  True
12   True                  True
13   True                  True
15   True                  True
16   True                  True
18   True                  True
19   True                  True
21   True                  True
22   True                  True
24   True                  True
25   True                  True
27   True                  True
28   True                  True
30   True                  True
31   True                  True
33   True                  True
34   True                  True
36   True                  True
37   True                  True
39   True                  True
40   True                  True
42   True                  True
43   True                  True
45   True                  True
46   True                  True
48   True                  True
49   True                  True
51   True                  True
52   True                  True
54   True                  True
55   True                  True
57   True                  True
58   True                  True
60   True                  True
61   True                  True
63   True                  True
64   True                  True
66   True                  True
67   True                  True
69   True                  True
70   True                  True
72   True                  True
73   True                  True
75   True                  True
76   True                  True
78   True                  True
79   True                  True
81   True                  True
82   True                  True
84   True                  True
85   True                  True
87   True                  True
88   True                  True
90   True                  True
91   True                  True
93   True                  True
94   True                  True
96   True                  True
97   True                  True
99   True                 False
100  True                 False
102  True                  True
103  True                  True
105  True                  True
106  True                  True
108  True                  True
109  True                  True
111  True                  True
112  True                  True
114  True                  True
115  True                  True
117  True                  True
118  True                  True
120  True                  True
121  True                  True
123  True                  True
124  True                  True
126  True                  True
127  True                  True
129  True                  True
130  True                  True
132  True                  True
133  True                  True
135  True                  True
136  True                  True
138  True                  True
139  True                  True
141  True                  True
142  True                  True
144  True                  True
145  True                  True
147  True                  True
148  True                  True
150  True                  True
151  True                  True
153  True                  True
154  True                  True
156  True                  True
157  True                  True
159  True                  True
160  True                  True
162  True                  True
163  True                  True
165  True                  True
166  True                  True
168  True                  True
169  True                  True
171  True                  True
172  True                  True
174  True                  True
175  True                  True
177  True                  True
178  True                  True
180  True                  True
181  True                  True
183  True                  True
184  True                  True
186  

In [32]:
combine_yr_ewc3 = uk_industry_total[['Year', 'EWC-STAT description']] == uk_managment_total[['Year', 'EWC-STAT description']]
combine_yr_ewc3

Year  EWC-STAT description
2    True                  True
5    True                  True
8    True                  True
11   True                  True
14   True                  True
17   True                  True
20   True                  True
23   True                  True
26   True                  True
29   True                  True
32   True                  True
35   True                  True
38   True                  True
41   True                  True
44   True                  True
47   True                  True
50   True                  True
53   True                  True
56   True                  True
59   True                  True
62   True                  True
65   True                  True
68   True                  True
71   True                  True
74   True                  True
77   True                  True
80   True                  True
83   True                  True
86   True                  True
89   True                  True
92   True                  True
95   True                  True
98   True                  True
101  True                 False
104  True                  True
107  True                  True
110  True                  True
113  True                  True
116  True                  True
119  True                  True
122  True                  True
125  True                  True
128  True                  True
131  True                  True
134  True                  True
137  True                  True
140  True                  True
143  True                  True
146  True                  True
149  True                  True
152  True                  True
155  True                  True
158  True                  True
161  True                  True
164  True                  True
167  True                  True
170  True                  True
173  True                  True
176  True                  True
179  True                  True
182  True                  True
185  True                  True
188  True                  True
191  True                  True
194  True                  True
197  True                  True
200  True                  True
203  True                 False
206  True                  True
209  True                  True
212  True                  True
215  True                  True
218  True                  True
221  True                  True
224  True                  True
227  True                  True
230  True                  True
233  True                  True
236  True                  True
239  True                  True
242  True                  True
245  True                  True
248  True                  True
251  True                  True
254  True                  True
257  True                  True
260  True                  True
263  True                  True
266  True                  True
269  True                  True
272  True                  True
275  True                  True
278  True                  True
281  True                  True
284  True                  True
287  True                  True
290  True                  True
293  True                  True
296  True                  True
299  True                  True
302  True                  True
305  True                 False
308  True                  True
311  True                  True
314  True                  True
317  True                  True
320  True                  True
323  True                  True
326  True                  True
329  True                  True
332  True                  True
335  True                  True
338  True                  True
341  True                  True
344  True                  True
347  True                  True
350  True                  True
353  True                  True
356  True                  True
359  True                  True
362  True                  True
365  True                  True
368  True                  True
371  True                  True
374  

In [33]:
combine_yr_ewc4 = uk_industry_type[['Year', 'EWC-STAT description']] == uk_managment_type[['Year', 'EWC-STAT description']]
combine_yr_ewc4

Year  EWC-STAT description
0    True                  True
1    True                  True
3    True                  True
4    True                  True
6    True                  True
7    True                  True
9    True                  True
10   True                  True
12   True                  True
13   True                  True
15   True                  True
16   True                  True
18   True                  True
19   True                  True
21   True                  True
22   True                  True
24   True                  True
25   True                  True
27   True                  True
28   True                  True
30   True                  True
31   True                  True
33   True                  True
34   True                  True
36   True                  True
37   True                  True
39   True                  True
40   True                  True
42   True                  True
43   True                  True
45   True                  True
46   True                  True
48   True                  True
49   True                  True
51   True                  True
52   True                  True
54   True                  True
55   True                  True
57   True                  True
58   True                  True
60   True                  True
61   True                  True
63   True                  True
64   True                  True
66   True                  True
67   True                  True
69   True                  True
70   True                  True
72   True                  True
73   True                  True
75   True                  True
76   True                  True
78   True                  True
79   True                  True
81   True                  True
82   True                  True
84   True                  True
85   True                  True
87   True                  True
88   True                  True
90   True                  True
91   True                  True
93   True                  True
94   True                  True
96   True                  True
97   True                  True
99   True                 False
100  True                 False
102  True                  True
103  True                  True
105  True                  True
106  True                  True
108  True                  True
109  True                  True
111  True                  True
112  True                  True
114  True                  True
115  True                  True
117  True                  True
118  True                  True
120  True                  True
121  True                  True
123  True                  True
124  True                  True
126  True                  True
127  True                  True
129  True                  True
130  True                  True
132  True                  True
133  True                  True
135  True                  True
136  True                  True
138  True                  True
139  True                  True
141  True                  True
142  True                  True
144  True                  True
145  True                  True
147  True                  True
148  True                  True
150  True                  True
151  True                  True
153  True                  True
154  True                  True
156  True                  True
157  True                  True
159  True                  True
160  True                  True
162  True                  True
163  True                  True
165  True                  True
166  True                  True
168  True                  True
169  True                  True
171  True                  True
172  True                  True
174  True                  True
175  True                  True
177  True                  True
178  True                  True
180  True                  True
181  True                  True
183  True                  True
184  True                  True
186  

## Results ENG: 
- the years match but the EWC-STAT do not (101, 203, 305, 407)
- calling all the False (i.e. all the total waste from different years- ) from the column EWC to see aren't the same, so we can remove before we concatinate meaning combine two worksheets.
- if == True: it would return all rows where they DO match, which would be all except 101, 203, 305, 407
- if == False: it would return all rows where they DO NOT match, which would be 101, 203, 305, 407

In [34]:
# loc-> locating

In [35]:
combine_yr_ewc.loc[combine_yr_ewc['EWC-STAT description'] == False]

Year  EWC-STAT description
101  True                 False
203  True                 False
305  True                 False
407  True                 False

In [36]:
combine_yr_ewc2.loc[combine_yr_ewc2['EWC-STAT description'] == False]

Year  EWC-STAT description
99   True                 False
100  True                 False
201  True                 False
202  True                 False
303  True                 False
304  True                 False
405  True                 False
406  True                 False

In [37]:
combine_yr_ewc3.loc[combine_yr_ewc3['EWC-STAT description'] == False]

Year  EWC-STAT description
101  True                 False
203  True                 False
305  True                 False
407  True                 False

In [38]:
combine_yr_ewc4.loc[combine_yr_ewc4['EWC-STAT description'] == False]

Year  EWC-STAT description
99   True                 False
100  True                 False
201  True                 False
202  True                 False
303  True                 False
304  True                 False
405  True                 False
406  True                 False

In [39]:
# Printing the length before we remove 'Total waste generation'

In [40]:
print(len(england_industry_total))
len(england_managment_total)

136


136

In [41]:
print(len(england_managment_type))
len(england_industry_type)

272


272

In [42]:
print(len(uk_industry_total))
len(uk_managment_total)

136


136

In [43]:
print(len(uk_industry_type))
len(uk_managment_type)

272


272

In [44]:
# Row Manipulation: Calling all the rows to remove the rows that have 'Total waste generation'
# Double Checking to make sure: Printing the length after remove 'Total waste generation'

# Row Manipulation: Calling all the rows to remove the rows that have 'Total waste treated'
# Double Checking to make sure: Printing the length after remove 'Total waste treated'

In [45]:
england_industry_total = england_industry_total.loc[england_industry_total['EWC-STAT description'] != '*Total waste generation',:]
len(england_industry_total)

132

In [46]:
england_managment_total = england_managment_total.loc[england_managment_total['EWC-STAT description'] != '*Total waste treated',:]
len(england_managment_total)

132

In [47]:
england_industry_type = england_industry_type.loc[england_industry_type['EWC-STAT description'] != '*Total waste generation',:]
len(england_industry_type)

264

In [48]:
england_managment_type = england_managment_type.loc[england_managment_type['EWC-STAT description'] != '*Total waste treated',:]
len(england_managment_type)

264

In [49]:
uk_industry_total = uk_industry_total.loc[uk_industry_total['EWC-STAT description'] != '*Total waste generation',:]
len(uk_industry_total)

132

In [50]:
uk_managment_total = uk_managment_total.loc[uk_managment_total['EWC-STAT description'] != '*Total waste treated',:]
len(uk_managment_total)

132

In [51]:
uk_industry_type = uk_industry_type.loc[uk_industry_type['EWC-STAT description'] != '*Total waste generation',:]
len(uk_industry_type)

264

In [52]:
uk_managment_type = uk_managment_type.loc[uk_managment_type['EWC-STAT description'] != '*Total waste treated',:]
len(uk_managment_type)

264

In [53]:
#### Merge Worksheets

# Above lines, we made sure Year and EWC Swat were exact same before we concatenated datasets
# Now that both tables are set to True Bouleans -> meaning removed total waste gen and treated; can combine both worksheets now that the unique subject identifier are both aligned

In [54]:
total_full = pd.concat([england_industry_total, england_managment_total], axis=1)

In [55]:
type_full = pd.concat([england_industry_type, england_managment_type], axis=1)

In [56]:
total_full_uk = pd.concat([uk_industry_total, uk_managment_total], axis=1)

In [57]:
type_full_uk = pd.concat([uk_industry_type, uk_managment_type], axis=1)

In [58]:
# Printing the length after we have mereged the datasets 

In [59]:
len(total_full)

132

In [60]:
len(type_full)

264

In [61]:
len(total_full_uk)

132

In [62]:
len(type_full_uk)

264

In [63]:
# Calling the column names of the concated dataset
# Checking if both mereged corectly and have all the columns; either by .columns or .head()

In [64]:
total_full.columns

Index(['Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'A', 'B', 'C10-C12', 'C13-C15', 'C16',
       'C17_C18', 'C19', 'C20-C22', 'C23', 'C24_C25', 'C26-C30', 'C31-C33',
       'D', 'E36_E37_E39', 'F', 'G-U_X_G4677', 'EP_HH',
       'Total waste generation', 'Year', 'EWC-STAT code',
       'EWC-STAT description', 'Hazardous/Non-hazardous split',
       'Energy recovery', 'Incineration',
       'Recovery other than energy recovery - Except backfilling',
       'Recovery other than energy recovery - Backfilling',
       'Deposit onto or into land',
       'Land treatment and release into water bodies', 'd`'],
      dtype='object')

In [65]:
type_full.columns

Index(['Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'A', 'B', 'C10-C12', 'C13-C15', 'C16',
       'C17_C18', 'C19', 'C20-C22', 'C23', 'C24_C25', 'C26-C30', 'C31-C33',
       'D', 'E36_E37_E39', 'F', 'G-U_X_G4677', 'EP_HH',
       'Total waste generation', 'Year', 'EWC-STAT code',
       'EWC-STAT description', 'Hazardous/Non-hazardous split',
       'Energy recovery', 'Incineration',
       'Recovery other than energy recovery - Except backfilling',
       'Recovery other than energy recovery - Backfilling',
       'Deposit onto or into land',
       'Land treatment and release into water bodies', 'd`'],
      dtype='object')

In [66]:
total_full_uk.columns

Index(['Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'A', 'B', 'C10-C12', 'C13-C15', 'C16',
       'C17_C18', 'C19', 'C20-C22', 'C23', 'C24_C25', 'C26-C30', 'C31-C33',
       'D', 'E36_E37_E39', 'F', 'G-U_X_G4677', 'EP_HH',
       'Total waste generation', 'Year', 'EWC-STAT code',
       'EWC-STAT description', 'Hazardous/Non-hazardous split',
       'Energy recovery', 'Incineration',
       'Recovery other than energy recovery - Except backfilling',
       'Recovery other than energy recovery - Backfilling',
       'Deposit onto or into land',
       'Land treatment and release into water bodies',
       'Total waste sent to final treatment'],
      dtype='object')

In [67]:
type_full_uk.head()

Year EWC-STAT code             EWC-STAT description  \
0  2010          01.1                   Spent solvents   
1  2010          01.1                   Spent solvents   
3  2010          01.2  Acid, alkaline or saline wastes   
4  2010          01.2  Acid, alkaline or saline wastes   
6  2010          01.3                        Used oils   

  Hazardous/Non-hazardous split                   A     B             C10-C12  \
0                     Hazardous                   0     0  84.949460377175086   
1                 Non-hazardous                   z     z                   z   
3                     Hazardous                   0     0  2644.4743226227511   
4                 Non-hazardous                   0     0  3317.5710114636677   
6                     Hazardous  31991.637710338411  8117  6775.0993747785724   

              C13-C15                 C16             C17_C18  \
0  25.203928515981719  45.058659743050029  2825.0664151318888   
1                   z                   z                   z   
3  1.5892666699683162  14.767553508758073   6197.973549642561   
4  1.7255892771741481  14.948493639590309  6310.9308768064948   
6  61.164692899388086  103.23224401765513  3148.6178313723299   

                  C19             C20-C22                 C23  \
0  44.387391014146118  95629.608909440503  100.80269133001863   
1                   z                   z                   z   
3  6717.2483825640811  40596.861821551203  2618.8016642224352   
4  756.56013947096244  47032.859106318632  2199.5689556199331   
6  4156.5604770921364  42942.453721045713  15806.007097942909   

              C24_C25             C26-C30             C31-C33  \
0  819.01638583385864  1917.6401368516108  774.24452051088565   
1                   z                   z                   z   
3  21429.840663347066  7506.9886629570701  894.47481985331126   
4   15261.28678688404  3985.4565683863866  915.37775022787775   
6  18624.625173560333  16112.105266055161  3093.1974314469066   

                    D         E36_E37_E39  F         G-U_X_G4677  \
0  21.471970574509989  1.4512891253133389  0  14969.237744563499   
1                   z                   z  z                   z   
3  28.026658899350352  116.80273758014525  0  8427.7860227646815   
4  29.145736803357117  118.13551548226599  0  10906.979298373008   
6  24973.976901330025  21.422419701255947  0  104950.65356123385   

                EP_HH Total waste generation  Year EWC-STAT code  \
0                   0     117258.13950301244  2010          01.1   
1                   z                      z  2010          01.1   
3                   0     97195.636126183381  2010          01.2   
4                   0     90850.545828753413  2010          01.2   
6  7575.3179999999993     288453.07190281467  2010          01.3   

              EWC-STAT description Hazardous/Non-hazardous split  \
0                   Spent solvents                     Hazardous   
1                   Spent solvents                 Non-hazardous   
3  Acid, alkaline or saline wastes                     Hazardous   
4  Acid, alkaline or saline wastes                 Non-hazardous   
6                        Used oils                     Hazardous   

  Energy recovery Incineration  \
0            8581            0   
1               z            z   
3               0            0   
4               0            0   
6               7            0   

  Recovery other than energy recovery - Except backfilling  \
0                                 8436.4869999999992         
1                                                  z         
3                                 79.013999999999996         
4                                 2863.9250000000002         
6                                 6417.9993996169451         

  Recovery other than energy recovery - Backfilling Deposit onto or into land  \
0                                                 0                         0   
1                          

In [68]:
# Exporting the file as csv

In [69]:
total_full.to_csv('England_Total_Full.csv')
type_full.to_csv('England_Type_Full.csv')
total_full_uk.to_csv('UK_Total_Full.csv')
type_full_uk.to_csv('UK_Type_Full.csv')

________

# 2nd Data Wrangle

# *UK & England*

## Load Type Data without a 'z'

- Making a list of missing value types

- Dropping "z"s from type without futher wrangle

In [70]:
missing_values = ["z"]
uk_type = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/UK_Type_Full.csv", na_values = missing_values)

In [71]:
uk_type.head()

Unnamed: 0  Year EWC-STAT code             EWC-STAT description  \
0           0  2010          01.1                   Spent solvents   
1           1  2010          01.1                   Spent solvents   
2           3  2010          01.2  Acid, alkaline or saline wastes   
3           4  2010          01.2  Acid, alkaline or saline wastes   
4           6  2010          01.3                        Used oils   

  Hazardous/Non-hazardous split            A       B      C10-C12    C13-C15  \
0                     Hazardous      0.00000     0.0    84.949460  25.203929   
1                 Non-hazardous          NaN     NaN          NaN        NaN   
2                     Hazardous      0.00000     0.0  2644.474323   1.589267   
3                 Non-hazardous      0.00000     0.0  3317.571011   1.725589   
4                     Hazardous  31991.63771  8117.0  6775.099375  61.164693   

          C16      C17_C18          C19       C20-C22           C23  \
0   45.058660  2825.066415    44.387391  95629.608909    100.802691   
1         NaN          NaN          NaN           NaN           NaN   
2   14.767554  6197.973550  6717.248383  40596.861822   2618.801664   
3   14.948494  6310.930877   756.560139  47032.859106   2199.568956   
4  103.232244  3148.617831  4156.560477  42942.453721  15806.007098   

        C24_C25       C26-C30      C31-C33             D  E36_E37_E39    F  \
0    819.016386   1917.640137   774.244521     21.471971     1.451289  0.0   
1           NaN           NaN          NaN           NaN          NaN  NaN   
2  21429.840663   7506.988663   894.474820     28.026659   116.802738  0.0   
3  15261.286787   3985.456568   915.377750     29.145737   118.135515  0.0   
4  18624.625174  16112.105266  3093.197431  24973.976901    21.422420  0.0   

     G-U_X_G4677     EP_HH  Total waste generation  Year.1 EWC-STAT code.1  \
0   14969.237745     0.000           117258.139503    2010            01.1   
1            NaN       NaN                     NaN    2010            01.1   
2    8427.786023     0.000            97195.636126    2010            01.2   
3   10906.979298     0.000            90850.545829    2010            01.2   
4  104950.653561  7575.318           288453.071903    2010            01.3   

            EWC-STAT description.1 Hazardous/Non-hazardous split.1  \
0                   Spent solvents                       Hazardous   
1                   Spent solvents                   Non-hazardous   
2  Acid, alkaline or saline wastes                       Hazardous   
3  Acid, alkaline or saline wastes                   Non-hazardous   
4                        Used oils                       Hazardous   

   Energy recovery  Incineration  \
0           8581.0           0.0   
1              NaN           NaN   
2              0.0           0.0   
3              0.0           0.0   
4              7.0           0.0   

   Recovery other than energy recovery - Except backfilling  \
0                                          8436.4870          
1                                                NaN          
2                                            79.0140          
3                                          2863.9250          
4                                          6417.9994          

   Recovery other than energy recovery - Backfilling  \
0                                                0.0   
1                                                NaN   
2                                                0.0   
3                                                0.0   
4                                                0.0   

   Deposit onto or into land  Land treatment and release into water bodies  \
0                   0.000000                                       0.00000   
1                        NaN                                           NaN   
2                 569.400000                                       0.00000   
3               73192.470000                                       0.0

In [72]:
uk_type.columns

Index(['Unnamed: 0', 'Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'A', 'B', 'C10-C12', 'C13-C15', 'C16',
       'C17_C18', 'C19', 'C20-C22', 'C23', 'C24_C25', 'C26-C30', 'C31-C33',
       'D', 'E36_E37_E39', 'F', 'G-U_X_G4677', 'EP_HH',
       'Total waste generation', 'Year.1', 'EWC-STAT code.1',
       'EWC-STAT description.1', 'Hazardous/Non-hazardous split.1',
       'Energy recovery', 'Incineration',
       'Recovery other than energy recovery - Except backfilling',
       'Recovery other than energy recovery - Backfilling',
       'Deposit onto or into land',
       'Land treatment and release into water bodies',
       'Total waste sent to final treatment'],
      dtype='object')

In [73]:
uk_total = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/UK_Total_Full.csv")

In [74]:
missing_values = ["z"]
eng_type = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/England_Type_Full.csv", na_values = missing_values)

In [75]:
eng_total = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/England_Total_Full.csv")

In [76]:
eng_total.columns

Index(['Unnamed: 0', 'Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'A', 'B', 'C10-C12', 'C13-C15', 'C16',
       'C17_C18', 'C19', 'C20-C22', 'C23', 'C24_C25', 'C26-C30', 'C31-C33',
       'D', 'E36_E37_E39', 'F', 'G-U_X_G4677', 'EP_HH',
       'Total waste generation', 'Year.1', 'EWC-STAT code.1',
       'EWC-STAT description.1', 'Hazardous/Non-hazardous split.1',
       'Energy recovery', 'Incineration',
       'Recovery other than energy recovery - Except backfilling',
       'Recovery other than energy recovery - Backfilling',
       'Deposit onto or into land',
       'Land treatment and release into water bodies', 'd`'],
      dtype='object')

#### Drop repeated/unwanted columns 

In [77]:
uk_type1 = uk_type.drop(['Unnamed: 0', 'EWC-STAT code', 'Year.1', 'EWC-STAT code.1', 'EWC-STAT description.1', 'Hazardous/Non-hazardous split.1'], axis=1) 

In [78]:
uk_type1.head(1)

Year EWC-STAT description Hazardous/Non-hazardous split    A    B  \
0  2010       Spent solvents                     Hazardous  0.0  0.0   

    C10-C12    C13-C15       C16      C17_C18        C19       C20-C22  \
0  84.94946  25.203929  45.05866  2825.066415  44.387391  95629.608909   

          C23     C24_C25      C26-C30     C31-C33          D  E36_E37_E39  \
0  100.802691  819.016386  1917.640137  774.244521  21.471971     1.451289   

     F   G-U_X_G4677  EP_HH  Total waste generation  Energy recovery  \
0  0.0  14969.237745    0.0           117258.139503           8581.0   

   Incineration  Recovery other than energy recovery - Except backfilling  \
0           0.0                                           8436.487          

   Recovery other than energy recovery - Backfilling  \
0                                                0.0   

   Deposit onto or into land  Land treatment and release into water bodies  \
0                        0.0                                           0.0   

   Total waste sent to final treatment  
0                            17017.487

In [79]:
uk_total1 = uk_total.drop(['Unnamed: 0', 'EWC-STAT code', 'Year.1', 'EWC-STAT code.1', 'EWC-STAT description.1', 'Hazardous/Non-hazardous split.1'], axis=1) 

In [80]:
uk_total1.head(1)

Year EWC-STAT description Hazardous/Non-hazardous split    A    B  \
0  2010       Spent solvents                         Total  0.0  0.0   

    C10-C12    C13-C15       C16      C17_C18        C19       C20-C22  \
0  84.94946  25.203929  45.05866  2825.066415  44.387391  95629.608909   

          C23     C24_C25      C26-C30     C31-C33          D  E36_E37_E39  \
0  100.802691  819.016386  1917.640137  774.244521  21.471971     1.451289   

     F   G-U_X_G4677  EP_HH  Total waste generation  Energy recovery  \
0  0.0  14969.237745    0.0           117258.139503           8581.0   

   Incineration  Recovery other than energy recovery - Except backfilling  \
0           0.0                                           8436.487          

   Recovery other than energy recovery - Backfilling  \
0                                                0.0   

   Deposit onto or into land  Land treatment and release into water bodies  \
0                        0.0                                           0.0   

   Total waste sent to final treatment  
0                            17017.487

In [81]:
eng_type1 = eng_type.drop(['Unnamed: 0', 'EWC-STAT code', 'Year.1', 'EWC-STAT code.1', 'EWC-STAT description.1', 'Hazardous/Non-hazardous split.1'], axis=1) 

In [82]:
eng_type1.head(1)

Year EWC-STAT description Hazardous/Non-hazardous split    A    B  \
0  2010       Spent solvents                     Hazardous  0.0  0.0   

     C10-C12   C13-C15        C16     C17_C18        C19       C20-C22  \
0  13.697857  9.133929  16.319214  693.591932  18.387391  39262.057248   

        C23     C24_C25     C26-C30   C31-C33         D  E36_E37_E39    F  \
0  9.554549  338.426256  355.766317  82.01455  2.012466     0.451289  0.0   

   G-U_X_G4677  EP_HH  Total waste generation  Energy recovery  Incineration  \
0  1046.579003    0.0               41847.992              0.0           0.0   

   Recovery other than energy recovery - Except backfilling  \
0                                           8436.487          

   Recovery other than energy recovery - Backfilling  \
0                                                0.0   

   Deposit onto or into land  Land treatment and release into water bodies  \
0                        0.0                                           0.0   

         d`  
0  8436.487

In [83]:
eng_total1 = eng_total.drop(['Unnamed: 0', 'EWC-STAT code', 'Year.1', 'EWC-STAT code.1', 'EWC-STAT description.1', 'Hazardous/Non-hazardous split.1'], axis=1) 

In [84]:
eng_total1.head(1)

Year EWC-STAT description Hazardous/Non-hazardous split    A    B  \
0  2010       Spent solvents                         Total  0.0  0.0   

     C10-C12   C13-C15        C16     C17_C18        C19       C20-C22  \
0  13.697857  9.133929  16.319214  693.591932  18.387391  39262.057248   

        C23     C24_C25     C26-C30   C31-C33         D  E36_E37_E39    F  \
0  9.554549  338.426256  355.766317  82.01455  2.012466     0.451289  0.0   

   G-U_X_G4677  EP_HH  Total waste generation  Energy recovery  Incineration  \
0  1046.579003    0.0               41847.992              0.0           0.0   

   Recovery other than energy recovery - Except backfilling  \
0                                           8436.487          

   Recovery other than energy recovery - Backfilling  \
0                                                0.0   

   Deposit onto or into land  Land treatment and release into water bodies  \
0                        0.0                                           0.0   

         d`  
0  8436.487

#### Renaming the columns 

In [85]:
uk_type1.rename(columns={'EWC-STAT description' : 'Waste_types', 'Hazardous/Non-hazardous split' : 'Haz_NonHaz', 'C10-C12' : 'C10_C12', 'C13-C15' : 'C13_C15', 'C20-C22' : 'C20_C22', 'C26-C30' : 'C26_C30', 'C31-C33' : 'C31_C33', 'G-U_X_G4677': 'G_U_X_G4677', 'Total waste generation' : 'Total_waste_gen', 'Energy recovery' : 'Energy_recovery', 'Recovery other than energy recovery - Except backfilling' : 'No_backfilling', 'Recovery other than energy recovery - Backfilling' : 'Backfilling', 'Deposit onto or into land' : 'Deposit_land', 'Land treatment and release into water bodies' : 'Treated_waterbodies', 'Total waste sent to final treatment' : 'Total_treatment'}, inplace=True)

In [86]:
uk_total1.rename(columns={'EWC-STAT description' : 'Waste_types', 'Hazardous/Non-hazardous split' : 'Haz_NonHaz', 'C10-C12' : 'C10_C12', 'C13-C15' : 'C13_C15', 'C20-C22' : 'C20_C22', 'C26-C30' : 'C26_C30', 'C31-C33' : 'C31_C33', 'G-U_X_G4677': 'G_U_X_G4677', 'Total waste generation' : 'Total_waste_gen', 'Energy recovery' : 'Energy_recovery', 'Recovery other than energy recovery - Except backfilling' : 'No_backfilling', 'Recovery other than energy recovery - Backfilling' : 'Backfilling', 'Deposit onto or into land' : 'Deposit_land', 'Land treatment and release into water bodies' : 'Treated_waterbodies', 'Total waste sent to final treatment' : 'Total_treatment'}, inplace=True)

In [87]:
uk_total1.columns

Index(['Year', 'Waste_types', 'Haz_NonHaz', 'A', 'B', 'C10_C12', 'C13_C15',
       'C16', 'C17_C18', 'C19', 'C20_C22', 'C23', 'C24_C25', 'C26_C30',
       'C31_C33', 'D', 'E36_E37_E39', 'F', 'G_U_X_G4677', 'EP_HH',
       'Total_waste_gen', 'Energy_recovery', 'Incineration', 'No_backfilling',
       'Backfilling', 'Deposit_land', 'Treated_waterbodies',
       'Total_treatment'],
      dtype='object')

In [88]:
eng_type1.rename(columns={'EWC-STAT description' : 'Waste_types', 'Hazardous/Non-hazardous split' : 'Haz_NonHaz', 'C10-C12' : 'C10_C12', 'C13-C15' : 'C13_C15', 'C20-C22' : 'C20_C22', 'C26-C30' : 'C26_C30', 'C31-C33' : 'C31_C33', 'G-U_X_G4677': 'G_U_X_G4677', 'Total waste generation' : 'Total_waste_gen', 'Energy recovery' : 'Energy_recovery', 'Recovery other than energy recovery - Except backfilling' : 'No_backfilling', 'Recovery other than energy recovery - Backfilling' : 'Backfilling', 'Deposit onto or into land' : 'Deposit_land', 'Land treatment and release into water bodies' : 'Treated_waterbodies', 'Total waste sent to final treatment' : 'Total_treatment'}, inplace=True)

In [89]:
eng_total1.rename(columns={'EWC-STAT description' : 'Waste_types', 'Hazardous/Non-hazardous split' : 'Haz_NonHaz', 'C10-C12' : 'C10_C12', 'C13-C15' : 'C13_C15', 'C20-C22' : 'C20_C22', 'C26-C30' : 'C26_C30', 'C31-C33' : 'C31_C33', 'G-U_X_G4677': 'G_U_X_G4677', 'Total waste generation' : 'Total_waste_gen', 'Energy recovery' : 'Energy_recovery', 'Recovery other than energy recovery - Except backfilling' : 'No_backfilling', 'Recovery other than energy recovery - Backfilling' : 'Backfilling', 'Deposit onto or into land' : 'Deposit_land', 'Land treatment and release into water bodies' : 'Treated_waterbodies', 'd`' : 'Total_treatment'}, inplace=True)

In [90]:
eng_total1.columns

Index(['Year', 'Waste_types', 'Haz_NonHaz', 'A', 'B', 'C10_C12', 'C13_C15',
       'C16', 'C17_C18', 'C19', 'C20_C22', 'C23', 'C24_C25', 'C26_C30',
       'C31_C33', 'D', 'E36_E37_E39', 'F', 'G_U_X_G4677', 'EP_HH',
       'Total_waste_gen', 'Energy_recovery', 'Incineration', 'No_backfilling',
       'Backfilling', 'Deposit_land', 'Treated_waterbodies',
       'Total_treatment'],
      dtype='object')

#### Export Datasets as csv 

In [91]:
uk_total1.to_csv('UK_Total_updated.csv')
uk_type1.to_csv('UK_Type_updated.csv')
eng_total1.to_csv('Eng_Total_updated.csv')
eng_type1.to_csv('Eng_Type_updated.csv')

_____

# 3rd Data Wrangle 

### Data Wrangling for all ANOVAs -> Reshaping the data by using melt function 

Devin's: reshaped_location = pd.melt(total_full.drop("EWC-STAT code", axis=1), id_vars=['Year','EWC-STAT description'], value_vars=total_full.iloc[:,3:20].columns,var_name = "Location_Type", value_name="Weight")

Meredith's: MeltedUKTotal1 = pd.melt(UK_Total_updated, id_vars = ['Year', 'Waste_types'], value_vars = ['A', 'B', 'C10_C12', 'C16'])

- This is to be able to find the ANOVAs of the different A,  B, C (classifications) columns. You will need to add all your other columns in

##### England: 

In [92]:
# Reshaping England Location(classification) by using melt() function

In [93]:
reshaped_location_eng = pd.melt(eng_total1, id_vars=['Year','Waste_types'], value_vars=eng_total1.iloc[:,3:20].columns, var_name = "Location_Type", value_name="Weight")

In [94]:
# Export file as csv 

In [95]:
reshaped_location_eng.to_csv("Reshaped_Location.csv", index=False)

In [96]:
# Reshaping England Treatment methods by using melt() function

In [97]:
reshaped_treatment_eng = pd.melt(eng_total1, id_vars=['Year','Waste_types'], value_vars=eng_total1.iloc[:,21:-1].columns,
       var_name = "Treatment_Type", value_name="Weight")

In [98]:
# Export file as csv 

In [99]:
reshaped_treatment_eng.to_csv("Reshaped_Treatment.csv", index=False)

In [100]:
# Grouping groupby() of waste and treatment by aggregate sum() function

In [101]:
total_waste_and_treatment_by_type_eng = eng_total1[["Waste_types", "Total_waste_gen", 
                                       "Total_treatment"]].groupby("Waste_types").sum().reset_index()

In [102]:
# Export file as csv 

In [103]:
total_waste_and_treatment_by_type_eng.to_csv("total_waste_and_treatment_by_type_Eng.csv", index=False)

##### UK: 

In [104]:
# Reshaping UK Location (classifications) by using melt() function 

In [105]:
reshaped_location_uk = pd.melt(uk_total1, id_vars=['Year', 'Waste_types'], value_vars=uk_total1.iloc[:,3:20].columns, var_name = "Location_Type", value_name="Weight")

In [106]:
# Export file as csv 

In [107]:
reshaped_location_uk.to_csv("UK_Reshaped_Location.csv", index=False)

In [108]:
# Reshaping UK Treatment methods by using melt() function 

In [109]:
reshaped_treatment_uk = pd.melt(uk_total1, id_vars=['Year', 'Waste_types'], value_vars=uk_total1.iloc[:,21:-1].columns,
        var_name = "Location_Type", value_name="Weight")

In [110]:
# Export file as csv 

In [111]:
reshaped_treatment_uk.to_csv("UK_Reshaped_Treatment.csv", index=False)

In [112]:
# Grouping groupby() of waste and treatment by aggregate sum() function 

In [113]:
total_waste_and_treatment_by_type_eng = uk_total1[["Waste_types", "Total_waste_gen", 
                                       "Total_treatment"]].groupby("Waste_types").sum().reset_index()

In [114]:
# Export file as csv 

In [115]:
total_waste_and_treatment_by_type_eng.to_csv("total_waste_and_treatment_by_type_UK.csv", index=False)

##### Combining 2 countires & Reshaping -> UK&England

In [116]:
uk_total1.columns

Index(['Year', 'Waste_types', 'Haz_NonHaz', 'A', 'B', 'C10_C12', 'C13_C15',
       'C16', 'C17_C18', 'C19', 'C20_C22', 'C23', 'C24_C25', 'C26_C30',
       'C31_C33', 'D', 'E36_E37_E39', 'F', 'G_U_X_G4677', 'EP_HH',
       'Total_waste_gen', 'Energy_recovery', 'Incineration', 'No_backfilling',
       'Backfilling', 'Deposit_land', 'Treated_waterbodies',
       'Total_treatment'],
      dtype='object')

In [117]:
eng_total11 = eng_total1.drop(['Total_waste_gen', 'Total_treatment'], axis=1)

In [118]:
uk_total11 = uk_total1.drop(['Year', 'Waste_types', 'Haz_NonHaz', 'Total_waste_gen', 'Total_treatment'], axis=1)

In [119]:
# Merging both countires eng and uk datasets 

In [120]:
countries = pd.concat([eng_total11, uk_total11], axis=1)

In [121]:
countries.columns

Index(['Year', 'Waste_types', 'Haz_NonHaz', 'A', 'B', 'C10_C12', 'C13_C15',
       'C16', 'C17_C18', 'C19', 'C20_C22', 'C23', 'C24_C25', 'C26_C30',
       'C31_C33', 'D', 'E36_E37_E39', 'F', 'G_U_X_G4677', 'EP_HH',
       'Energy_recovery', 'Incineration', 'No_backfilling', 'Backfilling',
       'Deposit_land', 'Treated_waterbodies', 'A', 'B', 'C10_C12', 'C13_C15',
       'C16', 'C17_C18', 'C19', 'C20_C22', 'C23', 'C24_C25', 'C26_C30',
       'C31_C33', 'D', 'E36_E37_E39', 'F', 'G_U_X_G4677', 'EP_HH',
       'Energy_recovery', 'Incineration', 'No_backfilling', 'Backfilling',
       'Deposit_land', 'Treated_waterbodies'],
      dtype='object')

In [122]:
# Reshaping by using melt() function 

In [123]:
reshaped_Countries = pd.melt(countries.drop('Haz_NonHaz', axis=1), id_vars=['Year','Waste_types'], value_vars=countries.iloc[:,3:-1].columns,
       var_name = "Location&Treatment_Type", value_name="Weight")

In [124]:
# Export file as csv 

In [125]:
reshaped_Countries.to_csv("Reshaped_Countries.csv", index=False)